### PR TITLE
fix: image previews on moonriver

### LINKED ISSUE #3338: Issue with indexedDB
### What happened?

It is possible that IndexedDB is failing because of either:
- Bad defined store
- no keys provided?

### Please reproduce in steps

- Open moonsama (https://localhost:9090/moonsama)
- see error in console.

However the I've encountered some IDB bug in Firefox 🤷 

### Expected Behavior

No IndexedDB error pls

### What browsers are you seeing the problem on?

_No response_

### Are you logged in?

No

### At which address did you encounter bug?

_No response_

### Screenshots

Screenshot 2022-07-06 at 12 15 20

### Relevant log output

```shell
index.js?8071:68 Uncaught (in promise) DOMException: Failed to execute 'get' on 'IDBObjectStore': No key or key range specified.
    at eval (webpack-internal:///./node_modules/idb-keyval/dist/index.js:83:100)
    at Array.map (<anonymous>)
    at eval (webpack-internal:///./node_modules/idb-keyval/dist/index.js:83:64)
    at eval (webpack-internal:///./node_modules/idb-keyval/dist/index.js:32:51)
```

### Payment link for reward

https://kodadot.xyz/transfer/?target=%3CMy_Kusama_Address_check_https://github.com/kodadot/nft-gallery/blob/main/CONTRIBUTING.md#creating-your-ksm-address%3E

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #3952: Replace main Pinata `gateway` with Kodadot pinata instance on IPFS
To fix the current 429 errors on various collections on MoonRiver, we need to start using https://kodadot.mypinata.cloud/

Probably related to https://github.com/kodadot/click

test here
https://beta.kodadot.xyz/movr/collection/0xd3a9c48df4d9342dc1a0ee2c185ce50588729fa9?page=1

`https://gateway.pinata.cloud/ipfs/QmUJLruvoTP2AmQy2KkC5tA2gfXif2hvUvehVmcAPjdhSH/461.png`

image

### LINKED ISSUE #4020: Image previews on Moonriver
Currently, there are only default placeholders being shown on for collections. appears on both landing and at explore/collections page, can we have. Can we have some previews? 
image
image

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #4020
- [x] PR closes #3952
- [x] PR closes #3338 

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
Screen Shot 2022-09-26 at 09 16 10
Screen Shot 2022-09-26 at 09 16 19

### CODE REVIEW

File: components/rmrk/utils.ts [hunk 1]
```diff
@@ -145,11 +145,43 @@ export const isArweaveUrl = (url: string): boolean => {
...
+  if (rr.test(ipfsUrl)) {
+    return ipfsUrl.replace('ipfs://', resolveProvider(provider))
+  }
+
+  const r = /^ipfs:\/\//
+  if (r.test(ipfsUrl)) {
+    return ipfsUrl.replace('ipfs://', `${resolveProvider(provider)}ipfs/`)
+  }
+
+  return sanitizeArweaveUrl(ipfsUrl, provider as ArweaveProviders)
```
[the remark this answers is not in the document]
Comment (preschian) [reply components/rmrk/utils.ts]: is this a bad thing?

**Comment by preschian** [comment]
some collection image was still using a placeholder because the metadata was `null`, maybe we need to fix that from the indexer?

and, I don't know how to fix this deepsource issue https://deepsource.io/gh/kodadot/nft-gallery/run/403d3893-71a4-4cf4-92a0-0882e1129d9a/javascript/JS-0336?listindex=1

**Comment by petersopko** [comment]
yeah wfm, except for this:

> some collection image was still using a placeholder because the metadata was `null`, maybe we need to fix that from the indexer?

perhaps related to this created issue 
- https://github.com/kodadot/click/issues/34

I think this is good solution until the indexer get's fixed?

**Comment by preschian** [comment]
> perhaps related to this created issue
> 
> * [How to handle collection metadata click#34](https://github.com/kodadot/click/issues/34)
> 
> I think this is good solution until the indexer get's fixed?

I see, yes I think so. I just realized this PR should fix this issue also https://github.com/kodadot/nft-gallery/issues/3338
that error occurs when we set `idb-keyval` with a `null` as a key, for example: `get(null, data)`

File: components/rmrk/utils.ts [hunk 2]
```diff
@@ -150,6 +150,10 @@ export const getSanitizer = (
   ipfsProvider?: ProviderKeyType,
   arProvider?: ArweaveProviders
 ): SanitizerFunc => {
+  if (url && url.includes('https://gateway.pinata.cloud')) {
+    return (link) => sanitizeIpfsUrl(link, ipfsProvider)
+  }
+
   if (isIpfsUrl(url)) {
     return (link) => sanitizeIpfsUrl(link, ipfsProvider)
   }
```
[the remark this answers is not in the document]
Comment (preschian) [reply components/rmrk/utils.ts:159]: oh, that was a good idea 👍

**Comment by petersopko** [comment]
@preschian let's check that deepsrouce issue and other than that, I think let's merge this to have some previews ready

**Review by petersopko — APPROVED** [review]
lgtm

**Review by Jarsen136 — APPROVED** [review]
code lgtm

**Comment by petersopko** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 45 USD/KSM ~ 1.111 $KSM
🧗 [DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM](https://kusama.subscan.io/account/DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)
🔗 [0x506e82136b2a226545f739f2ff511f82cbb70fb8083050ae2f1483e9a79ffd4c](https://kusama.subscan.io/extrinsic/0x506e82136b2a226545f739f2ff511f82cbb70fb8083050ae2f1483e9a79ffd4c)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by petersopko** [comment]
looks like auto-merge doesn't care about some of the deepsource issues which jumped out again after merging main into this, it doesn't look that serious, let's catch it elsewhere

**Pull request merged**